core/src/main/kotlin/io/customer/sdk/lifecycle/CustomerIOActivityLifecycleCallbacks.kt

```diff
@@ -0,0 +1,86 @@
+package io.customer.sdk.lifecycle
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import androidx.lifecycle.Lifecycle
+import io.customer.sdk.core.di.SDKComponent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.launch
+
+/**
+ * SDK component to listen to lifecycle events of all activities.
+ * The class is responsible for registering and unregistering the lifecycle callbacks.
+ * SDK clients can subscribe to lifecycle events of all activities using the [subscribe] function
+ * and should avoid listening to events from ActivityLifecycleCallbacks directly.
+ */
+class CustomerIOActivityLifecycleCallbacks : Application.ActivityLifecycleCallbacks {
```

Comment:
This is supposed to be the only `ActivityLifecycleCallbacks` listener in SDK, rest of the modules can listen to and use replay to get last event.